flang/test/Lower/intrinsic-procedures.f90

```diff
@@ -377,6 +377,28 @@ subroutine modulo_testi(r, a, p)
   r = modulo(a, p)
 end subroutine
 
+! SCAN 
+! CHECK-LABEL: func @_QPscan_test(%
+! CHECK-SAME: [[s:[^:]+]]: !fir.boxchar<1>, %
+! CHECK-SAME: [[ss:[^:]+]]: !fir.boxchar<1>) -> i32 
+integer function scan_test(s1, s2)
+  character(*) :: s1, s2
+! CHECK: %[[tmpBox:.*]] = fir.alloca !fir.box<!fir.heap<i32>>
+! CHECK-DAG: %[[c:.*]]:2 = fir.unboxchar %[[s]] : (!fir.boxchar<1>) -> (!fir.ref<!fir.char<1,?>>, index)
+! CHECK-DAG: %[[cBox:.*]] = fir.embox %[[c]]#0 typeparams %[[c]]#1 : (!fir.ref<!fir.char<1,?>>, index) -> !fir.box<!fir.char<1,?>>
+! CHECK-DAG: %[[cBoxNone:.*]] = fir.convert %[[cBox]] : (!fir.box<!fir.char<1,?>>) -> !fir.box<none>
+! CHECK-DAG: %[[c2:.*]]:2 = fir.unboxchar %[[ss]] : (!fir.boxchar<1>) -> (!fir.ref<!fir.char<1,?>>, index)
+! CHECK-DAG: %[[cBox2:.*]] = fir.embox %[[c2]]#0 typeparams %[[c2]]#1 : (!fir.ref<!fir.char<1,?>>, index) -> !fir.box<!fir.char<1,?>>
```

Comment:
Do all of these need to be CHECK-DAG? It looks like some of them have an ordering.

Reply:
Not sure. I based this on the trim example. I can try changing some of these.